src/main/java/net/minecraftforge/common/brewing/BrewingRecipeHelper.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2021.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.common.brewing;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.NonNullList;
+import net.minecraft.world.Container;
+import net.minecraft.world.Containers;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.RecipeManager;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BrewingStandBlockEntity;
+import net.minecraftforge.common.ForgeMod;
+import net.minecraftforge.event.ForgeEventFactory;
+import net.minecraftforge.fml.util.thread.EffectiveSide;
+import net.minecraftforge.fmllegacy.LogicalSidedProvider;
+
+import java.util.Optional;
+
+public class BrewingRecipeHelper
+{
+    /**
+     * Used by the brewing stand to determine if its contents can be brewed.
+     * Extra parameters exist to allow modders to create bigger brewing stands
+     * without much hassle
+     */
+    public static boolean canBrew(Level level, Container container, int reagentSlot, int[] baseSlots)
+    {
+        final ItemStack reagent = container.getItem(reagentSlot);
+        if (reagent.isEmpty()) return false;
+        final RecipeManager recipeManager = level.getRecipeManager();
+        for (int baseSlot : baseSlots)
+        {
+            ItemStack base = container.getItem(baseSlot);
+            final IBrewingContainer wrapper = new IBrewingContainer.Impl(base, reagent);
+            Optional<IBrewingRecipe> recipeFor = recipeManager.getRecipeFor(ForgeMod.BREWING.get(), wrapper, level);
+            if (recipeFor.isPresent()) return true;
+        }
+        return false;
+    }
+
+    /**
+     * Used by the brewing stand to brew its inventory Extra parameters exist to
```

Comment:
What are extra parameters? You are referring to the fact that `baseSlots` takes in an integer array to get all base slots to construct. Javadoc should be clearer.